internal/pproflogging/pproflogging.go

```diff
@@ -97,10 +103,228 @@ var pprofProfileRates = map[ProfileName]pprofSetRate{
 	},
 }
 
+// MaybeStartProfileBuffers start profile buffers for this process.
+func MaybeStartProfileBuffers(ctx context.Context) {
+	var err error
+
+	pcm, err := LoadProfileConfig(ctx, os.Getenv(EnvVarKopiaDebugPprof))
+	if err != nil {
+		log(ctx).With("error", err).Debug("cannot start configured profile buffers")
+		return
+	}
+
+	if pcm == nil {
+		log(ctx).Debug("no profile buffer configuration to start")
+		return
+	}
+
+	pprofConfigs.mu.Lock()
+	defer pprofConfigs.mu.Unlock()
+
+	pprofConfigs.pcm = pcm
+
+	log(ctx).Debug("no profile buffer configuration to start")
+
+	pprofConfigs.StartProfileBuffersLocked(ctx)
+}
+
+// MaybeRestartProfileBuffers used by SIGQUIT signal handlers to output PPROF data without exiting.
+func MaybeRestartProfileBuffers(ctx context.Context) {
+	pprofConfigs.mu.Lock()
```

Comment:
This would `panic` when `pprofConfigs == nil`.

Having said that, `pprofConfigs` is set to a non-nil value at package initialization time, so it should never be `nil` unless it is changed later at runtime.

Reply:
Correct.  If someone made the change than the function would fail, and the tests would fail, signaling to the developer that they made an error - which seems correct.  The code seems safe to me.